File: Microservice/model-categorizer-rust/src/classifiers/default_models.rs
use std::collections::HashSet;

/// Handles detection of default model configurations
pub struct DefaultModels {
    default_models: HashSet<String>,
}

impl DefaultModels {
    /// Creates a new default model detector
    pub fn new() -> Self {
        let models = [
            "gpt-3.5-turbo",
            "gpt-4",
            "gpt-4o",
            "claude-3-sonnet",
            "claude-3-opus",
            "gemini-1.5-pro",
            "gemini-1.5-flash",
            "gemini-2.0-pro",
        ];
        let default_models = models.iter().map(|s| s.to_string()).collect();
        DefaultModels { default_models }
    }

    /// Checks if a model ID corresponds to a default model
    pub fn is_default_model(&self, model_id: &str) -> bool {
        if self.default_models.contains(model_id) {
            return true;
        }
        let model_lower = model_id.to_lowercase();
        self.default_models
            .iter()
            .any(|m| m.eq_ignore_ascii_case(&model_lower))
    }
} 

File: Microservice/model-categorizer/node/classifiers/defaultModels.js
class DefaultModels {
  constructor() {
    this.defaultModels = {
      "gpt-3.5-turbo": true,
      "gpt-4": true,
      "gpt-4o": true,
      "claude-3-sonnet": true,
      "claude-3-opus": true,
      "gemini-1.5-pro": true,
      "gemini-1.5-flash": true,
      "gemini-2.0-pro": true,
    };
  }

  IsDefaultModel(modelID) {
    if (this.defaultModels[modelID]) return true;
    const lower = (modelID || "").toLowerCase();
    for (const key of Object.keys(this.defaultModels)) {
      if (lower === key.toLowerCase()) return true;
    }
    return false;
  }
}

module.exports = { DefaultModels };


